pay_platform_interface/lib/pay_platform_interface.dart

```diff
@@ -23,4 +23,7 @@ abstract class PayPlatform {
   Future<Map<String, dynamic>> showPaymentSelector(
       PaymentConfiguration paymentConfiguration,
       List<PaymentItem> paymentItems);
+
+  /// Update the payment status received from the backend server.
+  Future<void> updatePaymentStatus(bool isSuccess);
```

Comment:
how do you suggest I do it?

We can do it in multiple ways:
1. just handle it on iOS, and mention on the method's docs that this is iOS only.
2. handle it on iOS normally, but on Android handle it with an empty handler so nothing happens.
3. move it to the `pay_ios` package somehow, but then how do we make the APIs for Android and iOS the same?

Reply:
Starting from the assumption that this does not look like something that Google Pay may add, my suggestion is to create a child class that inherits from `PayMethodChannel` on `pay_ios`, and conditionally instantiate the regular or the iOS specific class in `pay.dart` from the `pay` package based on the platform.
What do you think?

Reply:
Starting from the assumption that this does not look like something that Google Pay may add, my suggestion is to create a child class that inherits from PayMethodChannel on pay_ios, and conditionally instantiate the regular or the iOS specific class in pay.dart from the pay package based on the platform.
What do you think?

Reply:
Starting from the assumption that this does not look like something that Google Pay may add, my suggestion is to create a child class that inherits from `PayMethodChannel` on `pay_ios`, and conditionally instantiate the regular or the iOS-specific class in `pay.dart` from the `pay` package based on the platform.
What do you think?

Reply:
> Hey @YazeedAlKhalaf, were you able to take a look at [this commet](https://github.com/google-pay/flutter-plugin/pull/200/files#r1257514591)?

I looked at it but forgot to comment my thoughts.

What about the public API for the update payment status, how will it be? Will the method always be there?

Reply:
So long as the method won't be used by other platforms, I'd skip #1 and #2, and avoid making this new method part of the interface signature.
I'd go ahead with #3 and #4. I feel that this would suffice with the iOS specific additional method.
What do you think?

Reply:
awesome will start with it, but the `pay/lib/src/pay.dart` will have the updatePaymentStatus method exposed, but on Android won't do anything, will probably check the type of the instantiated implementation

Reply:
i updated the code and test it, can you take a look @JlUgia?

Reply:
any updates? @JlUgia